src/Microsoft.ML.Core/Microsoft.ML.Core.csproj

```diff
@@ -6,11 +6,11 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <DefineConstants>CORECLR</DefineConstants>
     <RootNamespace>Microsoft.ML</RootNamespace>
+    <RunApiCompat>true</RunApiCompat>
```

Comment:
Can we instead set `RunApiCompat` based on if `IsStableProject` is set to true?

[https://github.com/dotnet/machinelearning/blob/master/build/BranchInfo.props#L2-L23](https://github.com/dotnet/machinelearning/blob/master/build/BranchInfo.props#L2-L23)

In the root Directory.Build.props, you should be able to say:

```xml
  <PropertyGroup>
    <RunApiCompat Condition="'$(RunApiCompat)` == '' AND '$(IsStableProject)' == 'true'">true</RunApiCompat>
  </PropertyGroup>
```

or simply:

```xml
  <PropertyGroup>
    <RunApiCompat Condition="'$(RunApiCompat)` == ''">$(IsStableProject)</RunApiCompat>
  </PropertyGroup>
```

Reply:
Sure that seems to represent more clearly the logic, I'll do it.

---
In reply to: [279941488](https://github.com/dotnet/machinelearning/pull/3623#discussion_r279941488) [](ancestors = 279941488)